frontend/src/components/custom-tools/output-analyzer-header/OutputAnalyzerHeader.jsx

```diff
@@ -3,29 +3,43 @@ import { ArrowLeftOutlined } from "@ant-design/icons";
 import { useNavigate, useParams } from "react-router-dom";
 
 import { useSessionStore } from "../../../store/session-store";
+import { useCustomToolStore } from "../../../store/custom-tool-store";
+
+let HeaderPublic;
+try {
+  HeaderPublic =
+    require("../../../plugins/prompt-studio-public-share/header-public/HeaderPublic.jsx").HeaderPublic;
+} catch (err) {
+  // Do nothing if plugins are not loaded.
+}
 
 function OutputAnalyzerHeader() {
   const { sessionDetails } = useSessionStore();
   const { id } = useParams();
   const navigate = useNavigate();
-
+  const { isPublicSource } = useCustomToolStore();
+  const handleNavigate = () => {
+    if (isPublicSource) {
+      navigate(-1);
```

Comment:
Incase few cases user might access directly through exact url in those cases there will be no previous route. So this will cause error. Always specify the exact route that need to be navigated.